### PR TITLE
feat: add auto-dbrp mapping and migration

### LINKED ISSUE #23457: Automate DBRP mappings
### Context
DBRP mappings have been around for a while but the experience of creating them hasn't been stellar and it's often not clear to users when they should be created and where to figure out how to do it. Additionally, we are bringing the InfluxQL REPL/shell back to v2 with the intention of making the on-ramp from v1 to v2 easier. That in mind, those users may create new buckets when starting their v2 experience and those will need to be mapped if they're to be used with this feature.

We want to automate this users don't have to do this ad hoc.

**Spec:**
-  Should have a migration that creates a mapping with `<bucketname>/autogen` for every bucket name where that makes sense.
- Should auto-create `<bucketname>/autogen` mappings for new buckets, unless the mapping already exists
- If bucketname has a slash, we autocreate a mapping instead based on the name

### PR DESCRIPTION
**Pull request by candrewlee14**
Closes #23457 

- Auto-map DBRPs when buckets are created or updated.
- Created migration to create a DBRP for each pre-existing bucket

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Rebased/mergeable
- [ ] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [x] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: dbrp/bucket_service.go [hunk 1]
```diff
@@ -37,13 +38,102 @@ func (s *BucketService) DeleteBucket(ctx context.Context, id platform.ID) error
...
+func ParseDBRP(bucketName string) (string, string) {
+	db, rp, isCut := strings.Cut(bucketName, "/")
+	if isCut {
+		return db, rp
+	}
+	return bucketName, "autogen"
+}
+
+// CreateBucket creates a new bucket and sets b.ID with the new identifier.
+// Also creates a corresponding auto-DBRP mapping.
```
Comment (lesam) [inline dbrp/bucket_service.go:62]: Do we need an explicitly created DBRP mapping if we can automate it?

I was thinking that for a db/rp pair, to find the bucket we would:
* first look at the table of mappings to see if there is a bucket there
* if not, see if there is a name match a la `ParseDBRP` above
* if not, 404

Comment (lesam) [reply dbrp/bucket_service.go:62]: If we do that we don't need to do much migration / extra creation.

Comment (lesam) [reply dbrp/bucket_service.go:62]: And it saves us from 'oh I deleted my dbrp mapping' tripping people up.

Comment (candrewlee14) [reply dbrp/bucket_service.go:62]: That probably would've been a better route to go, but I saw I could override the DBRP bucket service wrapper implementations, and knew I could create and update DBRPs there.
Do you think it's now worth reimplementing DBRPs virtually instead?

The Create, Update, and Delete bucket methods all handling DBRP mapping changes now makes it seem like a user wouldn't really need to bother with creating or deleting DBRPs manually either...

**Pull request closed without merging**